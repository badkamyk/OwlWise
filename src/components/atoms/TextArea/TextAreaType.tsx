import { InputProps } from "../Input/InputType";

export type TextAreaType = Omit<InputProps, "type"> & {
	rows?: number;
}