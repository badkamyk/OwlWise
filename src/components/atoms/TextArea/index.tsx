import { useFormContext } from "react-hook-form";
import InputLabel from "../InputLabel";
import { TextAreaType } from "./TextAreaType";

export default function TextArea({
	id,
	rows,
	placeholder,
	required,
}: TextAreaType) {
	const { register } = useFormContext();
	return (
		<div className="sm:col-span-2">
			<InputLabel text={"Your message"} htmlFor={id} />
			<textarea
				id={id}
				rows={rows}
				className="focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-500 dark:focus:border-primary-500 block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 shadow-sm dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400"
				placeholder={placeholder}
				required={required}
				{...register(id)}
			></textarea>
		</div>
	);
}
