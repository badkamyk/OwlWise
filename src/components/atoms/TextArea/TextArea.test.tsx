import { render, renderHook, screen } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import { expect, it } from "vitest";
import TextArea from "./index";

it("renders TextArea with props correctly", () => {
	const { result } = renderHook(() => useForm());
	const { container } = render(
		<FormProvider {...result.current}>
			<TextArea id="message" rows={4} placeholder="Your message" required />
		</FormProvider>
	);
	expect(screen.getByRole("textbox")).toBeTruthy();
	expect(screen.getByRole("textbox")).toHaveProperty("id", "message");
	expect(screen.getByRole("textbox")).toHaveProperty(
		"placeholder",
		"Your message"
	);
	expect(screen.getByRole("textbox")).toHaveProperty("required", true);
	expect(container.getElementsByClassName("sm:col-span-2")).toBeTruthy();
	expect(container).toMatchSnapshot();
});
