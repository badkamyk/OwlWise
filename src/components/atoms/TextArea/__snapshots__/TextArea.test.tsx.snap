// Vitest Snapshot v1

exports[`renders TextArea 1`] = `
<div>
  <div
    class="sm:col-span-2"
  >
    <label
      class="mb-2 block text-sm font-medium text-gray-900 dark:text-gray-300"
      for="message"
    >
      Your message
    </label>
    <textarea
      aria-invalid="false"
      class="focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-500 dark:focus:border-primary-500 block w-full rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-900 shadow-sm dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400"
      id="message"
      name="message"
      placeholder="Your message"
      required=""
      rows="4"
    />
  </div>
</div>
`;
